lib/info.js

```diff
@@ -0,0 +1,26 @@
+module.exports = class Info {
+  constructor (opts = {}) {
+    this.length = opts.length || 0
+    this.contiguousLength = opts.contiguousLength || 0
+    this.byteLength = opts.byteLength || 0
+    this.padding = opts.padding || 0
+  }
+
+  static empty () {
+    return new Info()
+  }
+
+  static async from (core, padding, snapshot) {
```

Comment:
I am curious why does this need to be async? is it out of anticipation for future information that might need awaiting opening a resource?

Reply:
> is it out of anticipation for future information that might need awaiting opening a resource?

Yep, we already have some ideas for additional information that will definitely result in IO.